disco_aws_automation/disco_elasticsearch.py

```diff
@@ -0,0 +1,404 @@
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "*"
+                    },
+                    "Action": "es:*",
+                    "Resource": resource,
+                    "Condition": {
+                        "IpAddress": {
+                            "aws:SourceIp": allowed_source_ips
+                        }
+                    }
+                }
+            ]
+        }
+        return json.dumps(policy)
+
+    def create(self, elasticsearch_name=None, es_config=None):
+        """
+        Create an ElasticSearch domain.
+        """
+        if elasticsearch_name:
+            desired_elasticsearch_names = [elasticsearch_name]
+        else:
+            desired_elasticsearch_names = self._get_elasticsearch_names()
+
+        all_elasticsearch_names = [domain_info["internal_name"] for domain_info in self.list()]
+
+        for elasticsearch_name in desired_elasticsearch_names:
+            domain_name = self.get_domain_name(elasticsearch_name)
+            if elasticsearch_name in all_elasticsearch_names:
+                logging.info('ElasticSearch domain %s already exists. Try updating it instead.', domain_name)
+                continue
+
+            logging.info('Creating ElasticSearch domain %s', domain_name)
+            # Get the latest elasticsearch cluster config.
+            desired_es_config = es_config or self._get_es_config(elasticsearch_name)
+            # Create a new elasticsearch config using the latest config.
+            throttled_call(self.conn.create_elasticsearch_domain, **desired_es_config)
+            self._add_route53(domain_name)
+
+    def update(self, elasticsearch_name=None, es_config=None):
+        """
+        Update an ElasticSearch domain.
+        """
+        if elasticsearch_name:
+            desired_elasticsearch_names = [elasticsearch_name]
+        else:
+            desired_elasticsearch_names = self._get_elasticsearch_names()
+
+        all_elasticsearch_names = [domain_info["internal_name"] for domain_info in self.list()]
+
+        for elasticsearch_name in desired_elasticsearch_names:
+            domain_name = self.get_domain_name(elasticsearch_name)
+            if elasticsearch_name not in all_elasticsearch_names:
+                logging.info('ElasticSearch domain %s does not exist. Try creating it instead.', domain_name)
+                continue
+            logging.info('Updating ElasticSearch domain %s', domain_name)
+            # Get the latest elasticsearch cluster config.
+            desired_es_config = es_config or self._get_es_config(elasticsearch_name)
+            # Update the elasticsearch cluster config to be the latest one.
+            throttled_call(self.conn.update_elasticsearch_domain_config, **desired_es_config)
+            self._add_route53(domain_name)
+
+    def delete(self, elasticsearch_name=None, delete_all=False):
+        """
+        Delete an ElasticSearch domain.
```

Comment:
As I understand it: If name is not passed in only ones in config are deleted. If someone creates a foobar domain it does not get deleted? While it somewhat counter intuitive its sane default, would however be helpful to document this.